### PR TITLE
Modify standard jobs to support use of Docker containers

### PR DESCRIPTION
**Pull request by git-harry**
Test job with Docker:
- https://rpc.jenkins.cit.rackspace.net/job/PM_rpc-gating-master-xenial-functional-test-gh/4

Test jobs with compute:
- https://rpc.jenkins.cit.rackspace.net/job/PM_rpc-gating-master-xenial-functional-test-gh/5
- https://rpc.jenkins.cit.rackspace.net/job/PM_rpc-openstack-master-xenial-swift-deploy-gh/1

Issue: [RE-1075](https://rpc-openstack.atlassian.net/browse/RE-1075)

### CODE REVIEW

**Comment by hughsaunders** [comment]
Please ignore unit test failure, I'm working on that.

**Comment by hughsaunders** [comment]
recheck_unit_tests

File: pipeline_steps/common.groovy [hunk 1]
```diff
@@ -684,6 +682,27 @@ void internal_slave(body){
   use_node("CentOS", body)
 }
 
+void standard_job_slave(String slave_type, body){
```
Comment (hughsaunders) [inline pipeline_steps/common.groovy]: body could be typed `Closure`

**Comment by hughsaunders** [comment]
Apologies for still broken unit test job.

File: pipeline_steps/common.groovy (hunk 1 above)
Comment (git-harry) [reply pipeline_steps/common.groovy]: Done

**Pull request merged**